src/exec/DB.hs

```diff
@@ -127,5 +127,6 @@ withDB dbOptions dbTracer indexTracer rr securityParam nodeConfig extLedgerState
         , cdbGcDelay = secondsToDiffTime 20
         }
   bracket (ChainDB.openDB chainDBArgs) ChainDB.closeDB $ \cdb ->
-    Sqlite.withIndexAuto epochSlots indexTracer (indexFilePath dbOptions) $ \idx ->
-      Index.trackChainDB rr idx cdb $ k idx cdb
+    Sqlite.withIndexAuto epochSlots indexTracer (indexFilePath dbOptions) $ \idx -> do
+      _ <- forkLinkedTransfer rr $ \rr' -> Index.trackChainDB rr' idx cdb
```

Comment:
Instead of `forkLinkedTransfer`, you can create a new resource registry for `Index.trackChainDB`as you will only use it for simple resources, not for threads. And as this is a scoped context you can use `with` and don't have to link to the parent registry.

So:
```haskell
ResourceRegistry.with $ \rr' -> Index.trackChainDB rr' idx cdb
```

Reply:
Of course, you had no way of knowing this, so I'll document this.